File: Semester 4/Programming and Projection Enironments/mpp-proiect-java/Proiect_MPP/AirportPersistence/src/main/java/root/persistence/people/IPersonRepository.java
package root.persistence.people;

import root.model.people.Person;
import root.persistence.Repository;

import java.util.Collection;

public interface IPersonRepository<T extends Person> extends Repository<T, Integer> {
    Collection<T> getPersonByFirstName(String firstName);
    Collection<T> getPersonByLastName(String lastName);
}

File: Semester 4/Programming and Projection Enironments/mpp-proiect-csharp/Proiect_MPP/persistence/people/IPersonRepository.cs
using model.people;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace persistence.people
{
    public interface IPersonRepository<T> : Repository<T, int> where T : Person
    {
        List<T> getPersonByFirstName(string firstName);
        List<T> getPersonByLastName(string lastName);
    }
}
